xdsl/dialects/transform.py

```diff
@@ -1,41 +1,325 @@
+    operand_def,
+    opt_attr_def,
+    opt_operand_def,
+    opt_prop_def,
+    prop_def,
+    region_def,
+    result_def,
+    var_operand_def,
+    var_result_def,
 )
+from xdsl.traits import IsolatedFromAbove, IsTerminator
+from xdsl.utils.str_enum import StrEnum
+
+
+class TransformHandleType(ParametrizedAttribute, TypeAttribute, ABC):
+    name: ClassVar[str] = field(init=False, repr=False)
 
 
 @irdl_attr_definition
-class AffineMapType(ParametrizedAttribute, TypeAttribute):
+class AffineMapType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#affinemapparamtype
+    """
+
     name = "transform.affine_map"
 
 
 @irdl_attr_definition
-class AnyOpType(ParametrizedAttribute, TypeAttribute):
+class AnyOpType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#anyoptype
+    """
+
     name = "transform.any_op"
 
 
 @irdl_attr_definition
-class AnyParamType(ParametrizedAttribute, TypeAttribute):
+class AnyParamType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#anyparamtype
+    """
+
     name = "transform.any_param"
 
 
 @irdl_attr_definition
-class AnyValueType(ParametrizedAttribute, TypeAttribute):
+class AnyValueType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#anyvaluetype
+    """
+
     name = "transform.any_value"
 
 
+@irdl_attr_definition
+class OperationType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#operationtype
+    """
+
+    name = "transform.op"
+    operation: ParameterDef[StringAttr]
+
+    def __init__(self, operation: str):
+        super().__init__(parameters=[StringAttr(operation)])
+
+
+@irdl_attr_definition
+class ParamType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#paramtype
+    """
+
+    name = "transform.param"
+    type: ParameterDef[TypeAttribute]
+
+    def __init__(self, type: TypeAttribute):
+        super().__init__(parameters=[type])
+
+
+@irdl_attr_definition
+class TypeParamType(TransformHandleType):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#typeparamtype
+    """
+
+    name = "transform.type"
+
+
+class FailurePropagationModeType(StrEnum):
+    PROPAGATE = "propagate"
+    SUPPRESS = "suppress"
+
+
+@irdl_attr_definition
+class FailurePropagationModeAttr(EnumAttribute[FailurePropagationModeType]):
+    name = "transform.failures"
+
+
+@irdl_op_definition
+class YieldOp(IRDLOperation):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#transformyield-transformyieldop
+    """
+
+    name = "transform.yield"
+
+    traits = frozenset([IsTerminator()])
+
+
+@irdl_op_definition
+class SequenceOp(IRDLOperation):
+    """
+    https://mlir.llvm.org/docs/Dialects/Transform/#transformsequence-transformsequenceop
+    """
+
+    name = "transform.sequence"
+
+    # TODO: Find out how to also use the enum FailurePropagationModeAttr as well as AnyIntegerAttr
```

Comment:
What do you mean by this?

Reply:
In milt, both integers can be used, as well as an attribute-type specifically made for this. I needed to figure out how I can support both at the same time. but this is now fixed

Reply:
Then it would be good to remove the todo